feat(adding new conditions doc): added new alert conditions doc, edit… (#15700)
newrelic/docs-website: src/content/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions.mdx

  Learn more about custom incident descriptions in our [docs](/docs/alerts-applied-intelligence/new-relic-alerts/advanced-alerts/advanced-techniques/alert-custom-incident-descriptions).

</Collapser>

  <Collapser
    id="runbook"
    title="Add runbook url"
  >
If you'd like to link to a runbook, you can put the URL in the runbook URL field.
  </Collapser>
</CollapserGroup>

</Step>
</Steps>

## Edit or improve an existing alert condition [#edit-existing-alert-condition]

If you want to edit an alert condition you've already created, you can:

1. Go to [one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > **Alerts & AI**.
2. Select **Alert Conditions** in the left navigation.
3. Click on the alert condition you want to edit.

From there, you will be able to see all of the elements you set when creating your condition. You can edit specific elements of the alert condition by clicking the pencil in top right of each section. 

If you have questions about how to edit each section, refer back to `Create a new alert condition` section:

> * [Signal coverage](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#create-alert-condition)

Reviewer: were going to change this to be called "Signal history" instead of "signal Coverage

Reviewer 2: @homelessbirds - still seeing this read as Signal coverage?

* [Signal behavior](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#advanced-signal-settings)
* [Thresholds](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#thresholds)
* [Additional details](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#additional-details) 

> ### Signal Coverage [#signal-coverage]

Reviewer 2: Should be Signal History

> Signal coverage gives a closer look at the most recent signals being ingested by the condition. For Synthetics type conditions, a chart is shown with the 5 most recent `SyntheticChecks`. For all other types, a line chart showing the `NraiSignal` over time for the last few days is shown for easier analysis. This chart is similar to what is shown for an individual issue or incident.

Reviewer 2: Signal history displays the activity of the signals that the condition has been producing.

Author: @corinnekunze when you say signal, can you give me a concrete example? I just want to be able to better explain this in plain language.

> If a condition is disabled, you’ll see an empty state message in this section of the Overview. In the case where the condition is not receiving a signal, you will see an error message prompting you to adjust your condition settings.

Reviewer 2: Can we have another section here that can be linked to from the empty state chart? We'd like a section that can help users debug why they are seeing empty states and take them here.

```suggestion
If a condition is disabled, you’ll see an empty state message in this section of the Overview. In the case where the condition is not receiving a signal, you will see an empty state prompting you to adjust your condition settings.

#### Troubleshooting your signal history chart
There are a few reasons that you may see an empty signal history chart.
- The condition is disabled and not producing any signals.
- The condition is a NRQL query and the metric or event it is targeting is not instrumenting data.
- The entity targeted by either the NRQL query or the condition is not instrumenting any data.
```

---

> title: Create and edit alert conditions in the UI

Reviewer 3: this title could capture the essence of your doc a bit better. To me, your doc is about taking a basic alert and adding alert conditions to modify the frequency (or whatever) of that basic alert. The title almost implies that 'in the UI' is the big subject of the doc, but that doesn't come across. Rather, this doc reads like the "step 2" of creating an alert, where the other advanced config doc is like a "step 3". Maybe a title could be like, idk, 'Modify a basic alert with alert conditions' ... You could def think of a better title but something along those lines.

tags:
  - Alerts and applied intelligence
  - Alerts
  - Alert conditions
translate:
  - jp
metaDescription: "Use the conditions page to identify what triggers an alert policy's notification, starting with the product and type of metric or service."
redirects:
  - /docs/alerts-applied-intelligence/new-relic-alerts/get-started/your-first-nrql-condition/
  - /docs/alerts/alert-policies/configuring-alerts/managing-your-alerts 
freshnessValidatedDate: never
---
import alertsSetaQuery from 'images/alerts_screenshot-crop_set-a-query.webp'

import alertsFineTuneAlertSignals from 'images/alerts_screenshot-crop_fine-tune-alert-signals.webp'

import alertsSetaThresholdforanAlertCondition from 'images/alerts_screenshot-crop_set-a-threshold-for-an-alert-condition.webp'

import alertsNameYourAlertCondition from 'images/alerts_screenshot-crop_name-your-alert-condition.webp'

import alertsAlertConditionPage from 'images/alerts_screenshot-crop_alert-condition-page.webp'

> New Relic alerts are built with alert conditions. If you want to be notified of any unusual behavior in your stack, you'll first need to create an alert condition. An alert condition is essentially a container that you create to define what conditions need to be met before you're notified of any unusual behavior. 

Reviewer 3: I think this intro could use a bit more value prop before diving into what they are. Since this tutorial is designed for people already using alerts, maybe you can have a sentence or two that's like...

"You can apply different alert conditions to existing alerts so you're notified about yadda yadda yadda" I'm guessing alert conditions can:

- Limit how may alerts you get
- Add more nuance to circumstances that need notifications
- idk some other use case

Alternatively, you can introduce the doc by describing alert fatigue and how users can create "stronger" alerts by applying alert conditions, THEN dive into how alert conditions affect alerts.

Reviewer 3: Also, watch out for passive voice.

> New Relic alerts are built with alert conditions. If you want to be notified of any unusual behavior in your stack, you'll first need to create an alert condition. An alert condition is essentially a container that you create to define what conditions need to be met before you're notified of any unusual behavior. 

Reviewer 3: ```suggestion
Alert conditions can elevate the quality of your alerts. An alert condition contains certain parameters or thresholds that need to be met before you're notified of any unusual behavior. They can mitigate excessive alerting, or let your team know when a new or unusual behavior appears
```

I actually think you can write something better but this is just an example of making it more active/direct w/ a story.

<img
    title="alert condition homepage"
    alt="A screenshot showing the alert condition homepage"
    src={alertsAlertConditionPage}
/>
<figcaption>
To use New Relic's universal alert condition experience, go to [one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > **Alerts & AI**, select **Alert Conditions** in the left navigation.
</figcaption>

With our **Alert Conditions** page you can:
* Browse all of your alert conditions 
* Easily identify any open issues
* Open existing alert conditions to edit thresholds or signal behavior 
* Create a new alert condition 

> This doc will walk you through how to edit and change [an _existing_ alert condition](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#edit-existing-alert-condition), and creating [new_ alert condition_](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#create-alert-condition]). 

Reviewer 3: ```suggestion
This doc will walk you through how to edit [an _existing_ alert condition](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#edit-existing-alert-condition) and how to create a [new_ alert condition_](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#create-alert-condition]). 
```

Reviewer 3: maybe we can further tighten the language with "This doc walks you through how" -- since it's a doc that exists, we can default to present tense

## Create a new alert condition [#create-alert-condition]

> For this example, we are going to show you how to manually create a new alert condition using the **Alert Condition** UI. But, if you're just beginning to use New Relic, we suggest creating an alert condition from a chart. To get started with alerts, review our [tutorial about creating your first alert](/docs/tutorial-create-alerts/create-an-alert/). 

Reviewer 3: ```suggestion
This example shows you how to manually create a new alert condition using the **Alert Condition** UI. If you're new to New Relic or haven't created an alert before, we recommend creating an alert condition from a chart. To get started with alerts, review our [tutorial about creating your first alert](/docs/tutorial-create-alerts/create-an-alert/). 
```

<Steps>

> <Step>

Reviewer 3: highlighting this needs a title; see below comment for recommendation

An alert condition is essentially a container that you create to define what conditions need to be met before you're notified of any unusual behavior.

<img
    title="set a query"
    alt="A screenshot showing a user how to set a query for an alert condition"
    src={alertsSetaQuery}
/>

<figcaption>
To create a new alert, go to [one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > **Alerts & AI**, select **Alert Conditions** in the left navigation, then select **New alert condition**.
</figcaption>

> <CollapserGroup>

Reviewer 3: I almost looked past this in the doc. This can be integrated into your first step, which can be titled 'Set your initial query' or even 'Define your initial query'

  <Collapser
    id="set-your-query"
    title="Set your query"
  >

> For this example, we will be using this query:

Reviewer 3: This can be unspooled a bit more to tell a story. I think having an intro line that describes the relationship between a NRQL query that retrieves data and how it can be "transformed" into an alert condition might be helpful.

```
SELECT average(duration) 
FROM PageView 
WHERE appName = 'WebPortal'
```
Using this query for your alert condition tells New Relic that you want to know the average pageviews from your `WebPortal` application. Monitoring `pageviews` can help you look out for any latency issues in your application. 

You can learn more about how to use NRQL, New Relic's query language, see our [NRQL documentation](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/create-nrql-alert-conditions/). 

</Collapser>
</CollapserGroup>
</Step>

<Step>
## Fine-tune advanced signal settings [#advanced-signal-settings]

> New Relic constantly observes the data that streams from your applications and into our system. But not all applications send signals at the same frequency or cadence. Some events could send signals to our system once a minute while others could only report data to New Relic once every day. An alert condition is a **specific** container designed for a **specific** use case. When creating an alert condition this section is the most customizable for the data that you're evaluating. 

Reviewer 3: A lot of this is repetition from the intro. I'd consider moving some of this into the intro or cutting it altogether, and sticking closely to the story you've set up. 

Instead, you might reinforce the connective tissue between the last query presented in step 1 and why customers got to this advanced landing page. How do they go from the query builder to this page? What navigation do they need to do to get here? What exactly does 'advanced signal settings' mean? (for example, 'advanced signals take a general alert condition and sets parameters around the duration, method, and delay of an alert. this is where you adjust the frequency of yadda yadda

We're going to customize these advanced signal settings for our condition that is montioring `pageviews` in the `WebPortal` application. 

<img
    title="fine tune alert settings"
    alt="A screenshot depicting advanced signal settings"
    src={alertsFineTuneAlertSignals}
/>

> <CollapserGroup>

Reviewer 3: Before you jump into these collapsers, i'd add the next "scene" for your story. 

Scene 1: a user is concerned with page views, hence the initial query. 
Scene 2, fine tune your signal: What decision and subsequent action does a user name to make/do for this page to be valuable?

Reviewer 3: i think laying out the example of how this page might be used will add value to the different variables they can modify, as described in your collapsers

  </Collapser>

  <Collapser
    id="sliding-window"
    title="Use sliding window aggregation"
  >

  Throughout the day, data streams from your application into New Relic. Instead of evaluating that data immediately for incidents, alert conditions collect the data over a period of time known as the **aggregation window**. An additional delay allows for slower data points to arrive before the window is aggregated.

  Sliding windows are helpful when you need to smooth out "spiky" charts. One common use case is to use sliding windows to smooth line graphs that have a lot of variation over short periods of time in cases where the rolling aggregate is more important than aggregates from narrow windows of time.

  We recommend using our sliding window aggregation if you're not expecting to have a steady and consistent stream of data but are expecting some dips and spikes in data. 

  You can learn more about sliding window aggregation in [this NRQL tutorial](/docs/query-your-data/nrql-new-relic-query-language/nrql-query-tutorials/create-smoother-charts-with-sliding-windows/) or by watching this video.

>   <Video

Reviewer 3: honest i'd cut the videos and consider using bulleted lists that are skimmable for the collapsers. right now it's a lot of text and can be overwhelming or give the impression that it's very complex. if we can keep the structure simple, then customers might be more emboldened to try it out

    type="youtube"
    id="-5--8DZynFE"
  />

  </Collapser>
  
  <Collapser
    id="streaming-method"
    title="Streaming method"
  >

>   In general, we recommend using the **event flow** streaming method. This is best for data that comes into your system frequently and steadily. There are specific cases where **event timer** might be a better method to choose, but for your first alert we recommend our default, **event flow**. To better understand which streaming method to choose, we recommend watching this brief video.  

Reviewer 3: I see that you're building story inside the collapser. I think it's good content, but per last comment, the section needs a bit more "set up" before you get into the meaty stuff here

  </Collapser>

  <Collapser
    id="gap-filling-strategy"
    title="Gap-filling strategy"
  >
  We're creating an alert condition to notify our team of any latency issues with the `WebPortal` application. In this example, our application consistently sends New Relic data. There is a constant stream of signals being sent from our application to New Relic and there is no expected gap in signal so we won't need to select a gap-filling strategy. We generally recommend leaving the **gap-filling strategy** set to **none**. 

  If you have a more inconsistent data set sending signals to New Relic once every twenty-four hours, then we'd recommend customizing this feature based on your team's specific needs. 

  Learn more about gap-filling strategies in our [lost signal docs](/docs/apis/nerdgraph/examples/nerdgraph-api-loss-signal-gap-filling/).

  </Collapser>
</CollapserGroup>

</Step>

<Step>

> ### Set thresholds for alert conditions [#thresholds]

Reviewer 3: ```suggestion
## Set thresholds for alert conditions [#thresholds]
```

> If an alert condition is a container, then thresholds are the rules that each alert condition contains. As data streams into your system, the alert condition searches for any incidents of these rules. If the alert condition sees data coming in from your system that has met all the conditions you've set, then it will create an incident. An incident is a signal that something is off in your system and you should take a look. 

Reviewer 3: I'd use a different word than 'contains', since it's defining itself with the word. "...are the rules that each alert condition must follow" or "defines the logic that opens the container"? I'm not sure -- the container metaphor is a little hard for me to understand. when i read 'container' in a docs context, my mind goes to docker containers

  </Collapser>
  
  <Collapser
    id="lost-signal"
    title="Lost signal threshold (optional)"
  >
Sometimes an incoming signal can be lost, and it's important to understand if it's simply a delay or an indication of a wider problem. Our **loss signal threshold** lets us know how many seconds the system should wait from the time the last data point was detected before considering the signal to be lost. If the signal does not return before the time limit that you set, you can choose one or both of two actions: open a new incident, and/or close all open incidents. You can close any related open incidents if the lost signal supersedes other incidents on this entity, or if the signal loss was expected.

Setting your lost signal threshold requires knowledge of your system and what you're looking to understand. In the case of `pageviews`, let's say New Relic collects a signal every minute. A lost signal could indicate a much larger latency issue. So, we recommend setting the time to your comfort level and then checking the box to **open a new lost signal incident**.
  </Collapser>
</CollapserGroup>
</Step>

<Step>

> ## Additional details [#additional-details]

Reviewer 3: I'd make this an action like the other steps, or break it out into its own h2

If you want to edit an alert condition you've already created, you can:

1. Go to [one.newrelic.com > All capabilities](https://one.newrelic.com/all-capabilities) > **Alerts & AI**.
2. Select **Alert Conditions** in the left navigation.
3. Click on the alert condition you want to edit.

From there, you will be able to see all of the elements you set when creating your condition. You can edit specific elements of the alert condition by clicking the pencil in top right of each section. 

If you have questions about how to edit each section, refer back to `Create a new alert condition` section:

* [Signal coverage](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#create-alert-condition)
* [Signal behavior](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#advanced-signal-settings)
* [Thresholds](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#thresholds)
* [Additional details](/docs/alerts-applied-intelligence/new-relic-alerts/alert-conditions/view-alert-conditions/#additional-details) 

> ### Signal Coverage [#signal-coverage]

Reviewer 3: Consider making this an action. Signal history/coverage doesn't immediately convey what a user might want to do (so for a user, they're thinking, 'How do I view my most recent alerts?' not 'How do I find my signal history')

> ### Signal Coverage [#signal-coverage]

Reviewer 3: hm, could this be an h2 instead?

> Signal coverage gives a closer look at the most recent signals being ingested by the condition. For Synthetics type conditions, a chart is shown with the 5 most recent `SyntheticChecks`. For all other types, a line chart showing the `NraiSignal` over time for the last few days is shown for easier analysis. This chart is similar to what is shown for an individual issue or incident.

Reviewer 3: what does it mean for a condition to ingest signals?

> If a condition is disabled, you’ll see an empty state message in this section of the Overview. In the case where the condition is not receiving a signal, you will see an empty state prompting you to adjust your condition settings.

Reviewer 3: Overview should be lower case but I think it needs a noun. Overview section? overview component? overview page?

> #### Troubleshooting your signal history chart

Reviewer 3: ```suggestion
#### Troubleshoot your signal history chart
```

Reviewer 3: I wonder if instead of an h4, we use a single collapser. i also think for every problem sentence, you should have a corresponding solution sentence, even if it seems self evident. For example, 

"...Enable the condition by yadda yadda"

There are a few reasons that you may see an empty signal history chart.
- The condition is disabled and not producing any signals.

> - The condition is a NRQL query and the metric or event it is targeting is not instrumenting data.

Reviewer 3: Do you mean that the Entity the query targets needs to be instrumented? Agents "instrument" an entity, and the NRQL query exposes the event/metric data captured by the agent, but the query itself doesn't instrument

- The entity targeted by either the NRQL query or the condition is not instrumenting any data.

> ## What's next

Reviewer 3: ```suggestion
## What's next? [#what's-next]
```

<DocTiles>

>   <DocTile title='Get started with New Relic alerts' path="/docs/tutorial-create-alerts/create-new-relic-alerts/"> A beginners guide to alerts</DocTile>

Reviewer 3: i like these tiles here. Maybe consider giving users more direction by defining the audience in the second sentence. That way they know which fits their use case
